src/View/Antlers/Language/Runtime/NodeProcessor.php

```diff
@@ -1681,7 +1682,9 @@ public function reduce($processNodes)
                         $this->data = $lockData;
 
                         if (is_object($output)) {
-                            if ($output instanceof Collection) {
+                            if ($output instanceof DataCollection) {
+                                $output = RuntimeValues::resolveWithRuntimeIsolation($output);
+                            } elseif ($output instanceof Collection) {
                                 $output = $output->all();
```

Comment:
You could probably just avoid checking for DataCollection and just `resolveWithRuntimeIsolation` for regular Collection instances too.

The test passes. I'm just not sure if/when you'd ever _not_ want to do it. 🤔 

```suggestion
                            if ($output instanceof Collection) {
                                $output = RuntimeValues::resolveWithRuntimeIsolation($output);
```

Reply:
I've made the change, and have added an extra sanity-check test in that file to ensure that tags returning `Collection` instances continue to work as intended.